server/controllers/handleQuestItemClicked.js

```diff
@@ -41,21 +41,20 @@ export const handleQuestItemClicked = async (req, res) => {
 
     const itemsCollectedByUser = world.dataObject?.keyAssets?.[keyAssetId]?.itemsCollectedByUser;
     const numberAllowedToCollect = keyAsset.dataObject?.numberAllowedToCollect;
-    let numberCollected = 0;
+    let numberCollectedToday = 1;
 
     if (
       itemsCollectedByUser &&
       itemsCollectedByUser[profileId] &&
-      itemsCollectedByUser[profileId][dateKey] &&
-      itemsCollectedByUser[profileId][dateKey].count >= numberAllowedToCollect
+      itemsCollectedByUser[profileId].totalCollectedToday >= numberAllowedToCollect
     ) {
       console.log(`Visitor has already collected ${numberAllowedToCollect} quest items today.`);
       return res.json({ addedClick: false, numberAllowedToCollect, success: true });
```

Comment:
The code is highly nested. I think we could remove this else, since we are already returning inside of the if statement